src/libraries/System.Private.CoreLib/src/System/HashCode.cs

```diff
@@ -319,6 +320,60 @@ public void AddBytes(ReadOnlySpan<byte> value)
             ref byte pos = ref MemoryMarshal.GetReference(value);
             ref byte end = ref Unsafe.Add(ref pos, value.Length);
 
+            if (Unsafe.ByteOffset(ref pos, ref end) < (sizeof(int) * 4))
+            {
+                goto Small;
+            }
+
+            // Usually Add calls Initialize but if we haven't used HashCode before it won't have been called.
+            if (_length == 0)
+            {
+                Initialize(out _v1, out _v2, out _v3, out _v4);
+            }
+            else
+            {
+                // If we have more than 16 bytes to hash, we can add them in 16-byte batches,
+                // but we first have to add enough data to flush any queued values.
+                switch (_length % 4)
+                {
+                    case 1:
+                        Debug.Assert(Unsafe.ByteOffset(ref pos, ref end) >= sizeof(int));
+                        Add(Unsafe.ReadUnaligned<int>(ref pos));
+                        pos = ref Unsafe.Add(ref pos, sizeof(int));
+                        goto case 2;
+                    case 2:
+                        Debug.Assert(Unsafe.ByteOffset(ref pos, ref end) >= sizeof(int));
+                        Add(Unsafe.ReadUnaligned<int>(ref pos));
+                        pos = ref Unsafe.Add(ref pos, sizeof(int));
+                        goto case 3;
+                    case 3:
+                        Debug.Assert(Unsafe.ByteOffset(ref pos, ref end) >= sizeof(int));
+                        Add(Unsafe.ReadUnaligned<int>(ref pos));
+                        pos = ref Unsafe.Add(ref pos, sizeof(int));
+                        break;
```

Comment:
This is inverted. It should be `case 3: /* code */ goto case 2; case 2: /* code */ goto case 1; case 1: /* code */ break;`

Reply:
If `_length % 4` is is 1/2/3, we need to increment it by 3/2/1 to make it divisible by 4.

Reply:
Ah right, I'm not awake enough to do math.